app/src/services/ShopService.ts

```diff
@@ -61,6 +63,9 @@ export type ReservationRepositoryInterface = {
   fetchShopsReservations(shopIds: number[]): Promise<Reservation[]>
   fetchShopReservations(shopId: number): Promise<Reservation[]>
   fetchStylistReservationCounts(stylistIds: number[]): Promise<{ stylistId: number, reservationCount: number }[]>
+  fetchReservationsByDSS(shopId: number, gteDate?: Date, ltDate?: Date, stylistId?: number):
+  Promise<{reservation: Date, duration: number}[]>
```

Comment:
why not return 
```js
Promise<(Reservation & { duration: number })[]>
```
reservation: Date is kinda weird, you'd expect an object but it only has a date object.

Reply:
test